boards: mchp: Add option to switch GPIO bank voltage

albertofloyd opened the pull request:
Add KConfig board support to match board jumper settings
Failing to update bank selection will affect GPIOs tied to VTR3

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

ulfalizer commented on soc/arm/microchip_mec/mec1501/Kconfig.soc:
```diff
@@ -66,3 +66,9 @@ config SOC_MEC1501_EXT_32K_PARALLEL_CRYSTAL
+
+config SOC_MEC1501_VTR3_1_8V
+	bool "VTR3 power rail is tied to 1.8V"
+	default n
```
`bool` symbols implicitly default to `n`, so can remove this.

Generates nicer docs without the `default n` too ("Implicitly defaults to n" instead of "n is <bunch of dependencies>").

ulfalizer commented on soc/arm/microchip_mec/mec1501/Kconfig.soc:
```diff
@@ -66,3 +66,9 @@ config SOC_MEC1501_EXT_32K_PARALLEL_CRYSTAL
+	bool "VTR3 power rail is tied to 1.8V"
+	default n
+	help
+	  Set this is if VTR3 power sourcejumper in the board is changed.
```
Missing newline at end of line.

galak commented on boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig:
```diff
@@ -7,8 +7,9 @@
+CONFIG_SOC_MEC1501_VTR3_1_8V=y
 CONFIG_BOARD_MEC1501MODULAR_ASSY6885=y
-CONFIG_RTOS_TIMER=n
+CONFIG_RTOS_TIMER=y
```
This change to RTOS_TIMER isn't related.  Should be at least a separate commit if not PR.

albertofloyd replied:
Removed this from this PR, will enable RTOS driver into another PR that enables other drivers as well.

galak reviewed (dismissed):
Need to deal w/RTOS_TIMER change in a separate commit.

albertofloyd committed:
boards: mchp: Add option to switch GPIO bank voltage

Add KConfig board support to match board jumper settings
Failing to update bank selection will affect GPIOs tied to VTR3

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

albertofloyd commented:
@galak , @ioannisg  please re-visit

ulfalizer reviewed (approved):
Kconfig nits fixed.

andrewboie commented:
@galak can you look at this one again?
